source/common/upstream/upstream_impl.cc

```diff
@@ -1459,21 +1458,22 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(const HostVector& new_hosts,
   const bool dont_remove_healthy_hosts =
       health_checker_ != nullptr && !info()->drainConnectionsOnHostRemoval();
   if (!current_priority_hosts.empty() && dont_remove_healthy_hosts) {
-    for (auto i = current_priority_hosts.begin(); i != current_priority_hosts.end();) {
-      if (!((*i)->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC) ||
-            (*i)->healthFlagGet(Host::HealthFlag::FAILED_EDS_HEALTH))) {
-        if ((*i)->weight() > max_host_weight) {
-          max_host_weight = (*i)->weight();
-        }
-
-        final_hosts.push_back(*i);
-        updated_hosts[(*i)->address()->asString()] = *i;
-        (*i)->healthFlagSet(Host::HealthFlag::PENDING_DYNAMIC_REMOVAL);
-        i = current_priority_hosts.erase(i);
-      } else {
-        i++;
-      }
-    }
+    Containers::removeMatchingElements(
+        current_priority_hosts,
+        [&updated_hosts, &final_hosts, &max_host_weight](const HostSharedPtr& p) mutable {
```

Comment:
nit: s/p/host/ 

in general in this PR I see more one-letter variable names than typical, beyond (say) loop indexes. I can't find anything style-guide-violating though I thought there used to be such rules.

https://google.github.io/styleguide/cppguide.html#General_Naming_Rules is the closest thing I could find.

Reply:
so I agree, but I think the diff is easier to read if what was a loop index (i, here) stays a similar shape (p, here).

but I am no C++ style person :).

Reply:
I can live with what's there if you want some unasked for $0.02 :)

Reply:
htuch breaks the tie.